### PR TITLE
Automated package builds and uploads using Travis CI

### PR DESCRIPTION
**Pull request by astorije**
## What this configuration does

- Downloads Node.js v8 using NodeSource.
  ⚠️ Note that `nvm` cannot be used because Node has to be installed system-wide.

- Builds the package (`deb/*.deb` file) like mentioned in the README

- Runs a couple tests to make sure that the file is correctly created, and is not empty (size of 0) or unexpectedly big (limit set to 10MB for now, which is pleeenty above the current 4MB files).
  💅 Bonus: Test output mentions the actual size ahead of time. See [this build](https://travis-ci.org/thelounge/deb-lounge/builds/340064517#L461) for example.
  ❓ Any suggestions for additional tests? The possibilities are rather limited at the moment though...

- When pushing a tag on this repo, uploads the built file to both `thelounge/lounge` and `thelounge/deb-lounge` (this repo).
  ⚠️ For this to work, the tag created on this repo **must** match the tag on the main repo. No big deal because that's the natural thing to do, but just to keep in mind (I use the same convention [on the website repo](https://github.com/thelounge/thelounge.github.io/releases) to build the changelog).
  - On `thelounge/lounge`, this will upload the file on the existing release
  - On this repo, it will create a GitHub release named after this tag before attaching the file. I hesitated not to upload to `thelounge/deb-lounge` but in the past people have missed that the packages are not on this repo, so I went for both to be safe, since it's free to do so. I can be convinced not to do so however, and instead add a mention in the README that built packages aren't here.

## What this configuration does not do

- The main things I was trying to achieve were uploading the package to GitHub, and testing that the built package installs and runs The Lounge correctly. Uploading works 👌, but there is no way to do the latter for now:
  - The package uses `systemd`/`systemctl` to run The Lounge, which is not available in Ubuntu 14.04, version that Travis uses. I have been discussing with their support, it's in the roadmap, but not there yet. Apart from that, `dpkg -i deb/*deb` works fine, I just can't test that TL is running.
  - When people ask for 16.04 support, they usually reply to use their Docker service... which does not support `systemd` (at least not without a ton of efforts, and not even sure).

  So I'll keep this as-is for now (file upload is more important right now anyway), and as soon as we can jump on 16.04, I'll do the other part.

- I didn't care for setting the release content on this repo yet (which doesn't support Markdown anyway so that's pretty useless) as that wasn't my main goal.

- I am only testing with Node 8 because at the moment this configuration only tests that the package builds correctly. When I can test that it can be installed and run, it will be useful to run it against multiple versions of Node.

## Other stuff

- I haven't squashed yet, but I will. I kept it as-is because I was exchanging emails with Travis support, and because I was bouncing back and forth with different solutions. It took me _a while_ to get there...

- This PR currently contains commit from #13 as I was testing stuff. I'll remove it before merging.

- In `postinst` step, I now do not create a home folder for the user for The Lounge. When I could install the package, it was flagged as an error, because `/etc/lounge` requires sudo permissions. I realized that this means [running the user config](https://github.com/thelounge/deb-lounge#user) is currently broken on `master`, so I went with not creating a home folder altogether. I've been [doing this from the beginning on `ansible-lounge`](https://github.com/astorije/ansible-lounge/blob/master/tasks/main.yml#L36) and it works fine.

- Everything in the current configuration has been tested. After tagging commits on this PR, I could see the file being uploaded in the main repo and this one. This is so cool. (I of course deleted uploaded files and tags/releases on this repo afterwards)

### CODE REVIEW

File: build-package [hunk 1]
```diff
@@ -11,10 +11,10 @@ OUTDIR="$STARTDIR/deb"
 rm -rf "$DESTDIR"
 
 # Install the package itself
-npm install -g --prefix "$DESTDIR/usr" thelounge@2.5.0
+npm install --unsafe-perm -g --prefix "$DESTDIR/usr" thelounge@2.7.0
```
Comment (xPaw) [inline build-package]: Could we infer version from `debian/control` somehow?

Comment (astorije) [reply build-package]: Probably, but that belongs to a further pr :)

File: systemd/user.service [hunk 2]
```diff
@@ -3,7 +3,7 @@ Description=The Lounge IRC client
 
 [Service]
 Type=simple
-ExecStart=/usr/bin/lounge start
+ExecStart=THELOUNGE_HOME=~/.lounge /usr/bin/thelounge start
```
Comment (xPaw) [inline systemd/user.service]: Shouldn't home be ``/etc/lounge` here or the ENV removed?

File: build-package (hunk 1 above)
Comment (xPaw) [reply build-package]: You already do something similar with `DEBFILE` in tests it seems

File: systemd/user.service (hunk 2 above)
Comment (astorije) [reply systemd/user.service]: That's for the user service. Also this is a commit coming from #13 that I need to remove here

Comment (xPaw) [reply systemd/user.service]: This change doesn't make sense in that PR either, does it? Why are you changing it to home folder if it was in etc before.

Comment (astorije) [reply systemd/user.service]: Because it's the user service and not the system service. Let's talk about this in the other pr

Comment (astorije) [reply systemd/user.service]: Or I can open a new PR just for that. It's just outside the scope of this PR, but I needed the change to try making it work

Comment (astorije) [reply systemd/user.service]: I'll clean up both PRs tomorrow

**Pull request merged**